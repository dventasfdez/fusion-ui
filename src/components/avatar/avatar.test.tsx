import React from 'react';
import renderer from 'react-test-renderer';
import Avatar, {IAvatarProps} from './avatar';

const avatarTest = (props: IAvatarProps) => (
  <Avatar {...props}>
    <img className="avatar-test" src="https://www.w3schools.com/howto/img_avatar.png" alt="avatar-img" />
  </Avatar>
);

test('Avatar should render and match snapshot', () => {
  const component = renderer.create(avatarTest({badge: 8}));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Avatar with no children', () => {
  const component = renderer.create(<Avatar badge={8} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Avatar big should render and match snapshot', () => {
  const component = renderer.create(avatarTest({big: true, badge: 8}));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Avatar small should render and match snapshot', () => {
  const component = renderer.create(avatarTest({small: true, badge: 8}));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Avatar xsmall should render and match snapshot', () => {
  const component = renderer.create(avatarTest({xsmall: true, badge: 8}));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Avatar with title should render and match snapshot', () => {
  const component = renderer.create(avatarTest({badge: 8, title: 'Avatar'}));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Avatar with title and subtitle should render and match snapshot', () => {
  const component = renderer.create(avatarTest({badge: 8, title: 'Avatar', subtitle: 'Avatar subtitle'}));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Avatar with title and subtitle disabled should render and match snapshot', () => {
  const onClick = jest.fn();
  const component = renderer.create(
    avatarTest({badge: 8, title: 'Avatar', subtitle: 'Avatar subtitle', disabled: true, onClick: onClick})
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
