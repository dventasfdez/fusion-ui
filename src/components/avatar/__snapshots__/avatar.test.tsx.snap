// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar big should render and match snapshot 1`] = `
<div
  className="avatar-wrapper "
>
  <div
    className="avatar-container"
  >
    <img
      alt="avatar-img"
      className="avatar_big avatar-test"
      src="https://www.w3schools.com/howto/img_avatar.png"
    />
    8
  </div>
</div>
`;

exports[`Avatar should render and match snapshot 1`] = `
<div
  className="avatar-wrapper "
>
  <div
    className="avatar-container"
  >
    <img
      alt="avatar-img"
      className="avatar avatar-test"
      src="https://www.w3schools.com/howto/img_avatar.png"
    />
    8
  </div>
</div>
`;

exports[`Avatar small should render and match snapshot 1`] = `
<div
  className="avatar-wrapper "
>
  <div
    className="avatar-container"
  >
    <img
      alt="avatar-img"
      className="avatar_small avatar-test"
      src="https://www.w3schools.com/howto/img_avatar.png"
    />
    8
  </div>
</div>
`;

exports[`Avatar with no children 1`] = `
<div
  className="avatar-wrapper "
>
  <div
    className="avatar-container"
  >
    8
  </div>
</div>
`;

exports[`Avatar with title and subtitle disabled should render and match snapshot 1`] = `
<div
  className="avatar-wrapper_disabled "
>
  <div
    className="avatar-container"
  >
    <button
      className="avatar"
      type="button"
    >
      <img
        alt="avatar-img"
        className="avatar-test"
        src="https://www.w3schools.com/howto/img_avatar.png"
      />
    </button>
    8
  </div>
  <div
    className="avatar-text-wrapper"
  >
    <span
      className="avatar-title"
    >
      Avatar
    </span>
    <span
      className="avatar-subtitle"
    >
      Avatar subtitle
    </span>
  </div>
</div>
`;

exports[`Avatar with title and subtitle should render and match snapshot 1`] = `
<div
  className="avatar-wrapper "
>
  <div
    className="avatar-container"
  >
    <img
      alt="avatar-img"
      className="avatar avatar-test"
      src="https://www.w3schools.com/howto/img_avatar.png"
    />
    8
  </div>
  <div
    className="avatar-text-wrapper"
  >
    <span
      className="avatar-title"
    >
      Avatar
    </span>
    <span
      className="avatar-subtitle"
    >
      Avatar subtitle
    </span>
  </div>
</div>
`;

exports[`Avatar with title should render and match snapshot 1`] = `
<div
  className="avatar-wrapper "
>
  <div
    className="avatar-container"
  >
    <img
      alt="avatar-img"
      className="avatar avatar-test"
      src="https://www.w3schools.com/howto/img_avatar.png"
    />
    8
  </div>
  <span
    className="avatar-title"
  >
    Avatar
  </span>
</div>
`;

exports[`Avatar xsmall should render and match snapshot 1`] = `
<div
  className="avatar-wrapper "
>
  <div
    className="avatar-container"
  >
    <img
      alt="avatar-img"
      className="avatar_xsmall avatar-test"
      src="https://www.w3schools.com/howto/img_avatar.png"
    />
  </div>
</div>
`;
